ibc-core/ics02-client/types/src/events.rs

```diff
@@ -51,6 +52,38 @@ impl From<ClientIdAttribute> for abci::EventAttribute {
         (CLIENT_ID_ATTRIBUTE_KEY, attr.client_id.as_str()).into()
     }
 }
+impl TryFrom<abci::EventAttribute> for ClientIdAttribute {
+    type Error = ClientError;
+
+    fn try_from(value: abci::EventAttribute) -> Result<Self, Self::Error> {
+        value
+            .key_str()
+            .map(|key| {
+                if key != CLIENT_ID_ATTRIBUTE_KEY {
+                    return Err(ClientError::InvalidAttributeKey {
+                        attribute_key: key.to_string(),
+                    });
+                }
+                value
+                    .value_str()
+                    .map(|value| {
+                        Ok(ClientIdAttribute {
+                            client_id: ClientId::from_str(value).map_err(|_| {
+                                ClientError::InvalidAttributeValue {
+                                    attribute_value: value.to_string(),
+                                }
+                            })?,
+                        })
+                    })
+                    .map_err(|_| ClientError::InvalidAttributeValue {
+                        attribute_value: String::new(),
+                    })?
+            })
+            .map_err(|_| ClientError::InvalidAttributeKey {
+                attribute_key: String::new(),
+            })?
+    }
+}
```

Comment:
The usage of `map` here and in many of the other `TryFrom` impls is a bit convoluted: in this case the `value.value_str().map()` being called _**inside**_ the `value.key_str().map()` call when it really doesn't need to be in there. It has a negative effect on readability in my opinion. Here's a version of `try_from` that I think reads a bit easier:
```rust
fn try_from(value: abci::EventAttribute) -> Result<Self, Self::Error> {
    if let Ok(key_str) = value.key_str() {
        if key_str != CLIENT_ID_ATTRIBUTE_KEY {
            return Err(ClientError::InvalidAttributeKey {
                attribute_key: key_str.to_string(),
            });
        }
    } else {
        return Err(ClientError::InvalidAttributeKey {
            attribute_key: String::new(),
        });
    }

    value
        .value_str()
        .map(|val_str| {
            let client_id = ClientId::from_str(val_str).map_err(|_| {
                ClientError::InvalidAttributeValue {
                    attribute_value: val_str.to_string(),
                }
            })?;

            Ok(ClientIdAttribute { client_id })
        })
        .map_err(|_| ClientError::InvalidAttributeValue {
            attribute_value: String::new(),
        })?
}
```
Would you mind going over the other `TryFrom` impls and cleaning them up in a similar fashion? 🙏

Reply:
Thanks for the detailed review @seanchen1991  🙏, I will refactor accordingly.